gef-ui/server/services/__tests__/api.test.js

```diff
@@ -240,22 +242,108 @@ describe('deleteFacility()', () => {
   });
 
-  it('throws an error if there is an api error', async () => {
-    Axios.get.mockReturnValue(Promise.reject());
-    await expect(api.getCompaniesHouseDetails({ companyRegNumber: companiesHouseNumber, userToken })).rejects.toThrowError();
+  it('returns the correct error information if it is called without a registration number', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
   });
 
-  it('throws an appropriate error when given an invalid companiesHouseNumber', async () => {
-    await expect(api.getCompaniesHouseDetails({ companyRegNumber: 'invalid', userToken })).rejects.toThrowError('Invalid company house number');
+  it('returns the correct error information if it is called with a null registration number', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber: null, userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with a registration number that is the empty string', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber: '', userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with an invalid registration number', async () => {
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber: MOCK_COMPANY_REGISTRATION_NUMBERS.INVALID_TOO_SHORT, userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg: 'Enter a valid Companies House registration number',
+    });
+  });
+
+  it.each([
+    {
+      status: 400,
+      errMsg: 'Enter a valid Companies House registration number',
+    },
+    {
+      status: 404,
+      errMsg: 'No company matching the Companies House registration number entered was found',
+    },
+    {
+      status: 422,
+      errMsg: 'UKEF can only process applications from companies based in the UK',
+    },
+  ])('returns the correct error information if the request to Portal API returns a $status', async ({ status, errMsg }) => {
+    const axiosError = new AxiosError();
+    axiosError.response = {
+      status,
+    };
+    Axios.get.mockRejectedValueOnce(axiosError);
+
+    const response = await api.getCompanyByRegistrationNumber({ registrationNumber, userToken });
+
+    expect(response).toEqual({
+      errRef: 'regNumber',
+      errMsg,
+    });
+  });
+
+  it('rethrows the error if the request to Portal API throws an error with an unhandled status', async () => {
+    const axiosError = new AxiosError();
+    axiosError.response = {
+      status: 418,
```

Comment:
Should this be `418`?

Reply:
This test is just for an arbitrary unhandled status, but I agree the choice of `418` might be confusing so I've changed it to `500`.